core/node/node_framework/examples/main_node.rs

```diff
@@ -126,6 +137,95 @@ impl MainNodeBuilder {
         Ok(self)
     }
 
+    fn add_tx_sender_layer(mut self) -> anyhow::Result<Self> {
+        let state_keeper_config = StateKeeperConfig::from_env()?;
+        let rpc_config = ApiConfig::from_env()?.web3_json_rpc;
+        let network_config = NetworkConfig::from_env()?;
+        let postgres_storage_caches_config = PostgresStorageCachesConfig {
+            factory_deps_cache_size: rpc_config.factory_deps_cache_size() as u64,
+            initial_writes_cache_size: rpc_config.initial_writes_cache_size() as u64,
+            latest_values_cache_size: rpc_config.latest_values_cache_size() as u64,
+        };
+
+        // On main node we always use master pool sink.
+        self.node.add_layer(TxSinkLayer::MasterPoolSink);
+        self.node.add_layer(TxSenderLayer::new(
+            TxSenderConfig::new(
+                &state_keeper_config,
+                &rpc_config,
+                network_config.zksync_network_id,
+            ),
+            postgres_storage_caches_config,
+            rpc_config.vm_concurrency_limit(),
+            ApiContracts::load_from_disk(), // TODO (BFT-138): Allow to dynamically reload API contracts
+        ));
+        Ok(self)
+    }
+
+    fn add_http_web3_api_layer(mut self) -> anyhow::Result<Self> {
+        let rpc_config = ApiConfig::from_env()?.web3_json_rpc;
+        let contracts_config = ContractsConfig::from_env()?;
+        let network_config = NetworkConfig::from_env()?;
+        let state_keeper_config = StateKeeperConfig::from_env()?;
+        let with_debug_namespace = state_keeper_config.save_call_traces;
+
+        let mut namespaces = Namespace::DEFAULT.to_vec();
+        if with_debug_namespace {
+            namespaces.push(Namespace::Debug)
+        }
+        namespaces.push(Namespace::Snapshots);
+
+        let optional_config = Web3ServerOptionalConfig {
+            namespaces: Some(namespaces),
+            filters_limit: Some(rpc_config.filters_limit()),
+            subscriptions_limit: Some(rpc_config.subscriptions_limit()),
+            batch_request_size_limit: Some(rpc_config.max_batch_request_size()),
+            response_body_size_limit: Some(rpc_config.max_response_body_size()),
+            tree_api_url: rpc_config.tree_api_url(),
+            ..Default::default()
+        };
+        self.node.add_layer(Web3ServerLayer::http(
+            rpc_config.http_port,
+            InternalApiConfig::new(&network_config, &rpc_config, &contracts_config),
+            optional_config,
+        ));
+
+        Ok(self)
+    }
+
+    fn add_ws_web3_api_layer(mut self) -> anyhow::Result<Self> {
```

Comment:
It looks like with some effort, copy-paste between this method and `add_http_web3_api_layer` could be avoided, but I understand that it's not the focus rn.

Reply:
Yup, this will be worked on separately.